File: frontend/src/types/inventory.ts
export interface Inventory {
  inventory_id: number;
  brand_id: number;
  current_quantity: number;
  minimum_quantity: number;
  maximum_quantity: number;
  reorder_quantity: number;
  batch_number?: string;
  expiry_date?: string;
  manufacturing_date?: string;
  location?: string;
  status: InventoryStatus;
  last_restocked_at?: string;
  created_at: string;
  updated_at: string;
}

export interface InventoryWithDetails extends Inventory {
  brand_name: string;
  arabic_name?: string;
  generic_name: string;
  manufacturer_name?: string;
  dosage_form?: string;
}

export interface InventoryMovement {
  movement_id: number;
  inventory_id: number;
  movement_type: MovementType;
  quantity: number;
  previous_quantity: number;
  new_quantity: number;
  reference_type?: string;
  reference_id?: number;
  reason?: string;
  performed_by?: string;
  notes?: string;
  created_at: string;
}

export interface InventoryAlert {
  alert_id: number;
  inventory_id: number;
  alert_type: AlertType;
  threshold_value?: number;
  current_value?: number;
  severity: string;
  is_resolved: boolean;
  resolved_at?: string;
  resolved_by?: string;
  message?: string;
  created_at: string;
}

export interface InventoryStats {
  total_items: number;
  low_stock_items: number;
  out_of_stock_items: number;
  expiring_soon_items: number;
  expired_items: number;
  total_value?: number;
}

export type InventoryStatus = 'available' | 'low_stock' | 'out_of_stock' | 'expired' | 'expiring_soon' | 'damaged';
export type MovementType = 'add' | 'remove' | 'return' | 'damage' | 'sale' | 'adjustment' | 'transfer';
export type AlertType = 'low_stock' | 'expiry_soon' | 'expired' | 'overstock';

File: app/schemas/inventory.py
from pydantic import BaseModel, Field
from datetime import datetime
from typing import Optional, List
from app.core.constants import InventoryStatus, MovementType, AlertType


class InventoryBase(BaseModel):
    brand_id: int
    current_quantity: int = Field(default=0, ge=0)
    minimum_quantity: int = Field(default=10, ge=0)
    maximum_quantity: int = Field(default=1000, ge=0)
    reorder_quantity: int = Field(default=50, ge=0)
    batch_number: Optional[str] = None
    expiry_date: Optional[datetime] = None
    manufacturing_date: Optional[datetime] = None
    location: Optional[str] = None
    status: InventoryStatus = InventoryStatus.AVAILABLE


class InventoryCreate(InventoryBase):
    pass


class InventoryUpdate(BaseModel):
    current_quantity: Optional[int] = Field(default=None, ge=0)
    minimum_quantity: Optional[int] = Field(default=None, ge=0)
    maximum_quantity: Optional[int] = Field(default=None, ge=0)
    reorder_quantity: Optional[int] = Field(default=None, ge=0)
    batch_number: Optional[str] = None
    expiry_date: Optional[datetime] = None
    manufacturing_date: Optional[datetime] = None
    location: Optional[str] = None
    status: Optional[InventoryStatus] = None


class Inventory(InventoryBase):
    inventory_id: int
    last_restocked_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True


class InventoryWithDetails(Inventory):
    brand_name: str
    generic_name: str
    manufacturer_name: Optional[str] = None
    dosage_form: Optional[str] = None


class InventoryMovementBase(BaseModel):
    inventory_id: int
    movement_type: MovementType
    quantity: int = Field(..., gt=0)
    reference_type: Optional[str] = None
    reference_id: Optional[int] = None
    reason: Optional[str] = None
    notes: Optional[str] = None


class InventoryMovementCreate(InventoryMovementBase):
    pass


class InventoryMovement(InventoryMovementBase):
    movement_id: int
    previous_quantity: int
    new_quantity: int
    performed_by: Optional[str] = None
    created_at: datetime
    
    class Config:
        from_attributes = True


class InventoryAlertBase(BaseModel):
    inventory_id: int
    alert_type: AlertType
    threshold_value: Optional[int] = None
    current_value: Optional[int] = None
    severity: str = "MEDIUM"
    message: Optional[str] = None


class InventoryAlertCreate(InventoryAlertBase):
    pass


class InventoryAlert(InventoryAlertBase):
    alert_id: int
    is_resolved: bool = False
    resolved_at: Optional[datetime] = None
    resolved_by: Optional[str] = None
    created_at: datetime
    
    class Config:
        from_attributes = True


class InventoryStats(BaseModel):
    total_items: int
    low_stock_items: int
    out_of_stock_items: int
    expiring_soon_items: int
    expired_items: int
    total_value: Optional[float] = None
